src/main/java/org/springframework/data/couchbase/core/convert/DateConverters.java

```diff
@@ -52,9 +56,18 @@ private DateConverters() {
   public static Collection<Converter<?, ?>> getConvertersToRegister() {
     List<Converter<?, ?>> converters = new ArrayList<Converter<?, ?>>();
 
-    converters.add(DateToLongConverter.INSTANCE);
+    boolean useISOStringConverterForDate = Boolean.parseBoolean(
+            System.getProperty("org.springframework.data.couchbase.useISOStringConverterForDate", "false"));
+
+    if (useISOStringConverterForDate) {
+      converters.add(DateToStringConverter.INSTANCE);
+      converters.add(StringToDateConverter.INSTANCE);
+    } else {
+      converters.add(DateToLongConverter.INSTANCE);
+      converters.add(NumberToDateConverter.INSTANCE);
```

Comment:
I wonder if adding the `NumberToDateConverter` still in the case of the property being set to true would work? This would allow to migrate data that was serialized with a `long` encoding for dates to a `String` based format.
("Be liberal in what you accept, and strict in what you produce")